### ISSUE TITLE
[QST/FEA] Dask-cudf doesn't infer divisions

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/cudf#2292]
**Is your feature request related to a problem? Please describe.**
Currently certain operations in Dask like `set_index` sorts the Dataframe by index and creates clean partitions based on this index in such a way that all rows of a `unique` index belong to the same partition. It also infers the divisions of each partition.

Dask-cudf on the other hand has an implementation of `set_index` which does not infer the divisions of the dataframe. This is an issue downstream as I want to repartition my dataframe based on divisions.

**Describe the solution you'd like**
Would it be possible to infer divisions/ partition the dataframe in a clean way. This would potentially allow users to use `map_parititons` with the legacy groupby method with their own custom methods to implement any `groupby` aggregation.

**Additional context**
```
ddf = dask_cudf.read_csv(data_dir+'file*.csv')
ddf.divisions
(None, None, None)
ddf = ddf.set_index(index='col0')
ddf.divisions
(None, None, None)
```

**Comment by mrocklin** [comment]
Yes, if we use the mainline dask.dataframe set_index implementation then you should get everything that you want.  Additionally, operations like groupby-apply will see that this is the situation, and do the smart thing automatically.

There is some long-standing effort to accomplish this.  I think that currently it's waiting on a cudf searchsorted operation.  This is the same effort that would enable dask.dataframe's sort/join code to work with cudf dataframes.  They're all essentially the same operation under the hood.

**Comment by kkraus14** [comment]
👍 That's in progress for this release, and once that's in we can start playing whack a mole to try to get it working.

**Comment by kkraus14** [comment]
Fixed by #2632
